### PR TITLE
Add option to disable state verification

### LINKED ISSUE #1101: invalid_auth when using InstallProvider with no bot
### Description

I'm following the oauthv2 guide here: https://slack.dev/node-slack-sdk/oauth

However right now I do not have a bot user associated with the app and I'm not requesting a bot scope.

When calling `installer.handleCallback` the call to `getBotId` fails with `invalid_auth` as no bot token exists.

Should I not be using the installer for user only tokens?

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Packages:

Select all that apply:

- [ ] `@slack/web-api`
- [ ] `@slack/events-api`
- [ ] `@slack/interactive-messages`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [x] `@slack/oauth`
- [ ] I don't know

#### Reproducible in:

package version: 1.2.0

node version: v14.12.0

OS version(s): 10.15.7

#### Steps to reproduce:

1. Create oauthv2 flow with new app
2. Don't create a bot for the app, and for scopes / userScopes use:
```
    scopes: [],
    userScopes: ['users:read', 'chat:write'],
```
3. Run through "add to slack" flow
#### Expected result:

Can authenticate user

#### Actual result:

invalid_auth when the installer tries to get the bot id (no bot token exist)
Logs, screenshots, screencast, sample project, funny gif, etc.

### PR DESCRIPTION
**Pull request by srajiang**
###  Summary

This PR partially fixes [bolt-js/#1101](https://github.com/slackapi/bolt-js/issues/1101) by adding the option to disable state verification during OAuth flow via a `stateVerification` flag. By default `stateVerification = true`. Users can pass `stateVerification: false` as an OAuth option via App `installerOptions` object.

Previous (reverted) [changes](https://github.com/slackapi/node-slack-sdk/commit/b02f976a3d852a3d4b90ee6ce46fe64b327ade78) allowed option to disable state verification but continued usage of default state store for state generation. This current set of changes disables the default state store when stateVerification = false. 

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/oauth/src/index.ts [hunk 1]
```diff
@@ -41,13 +41,14 @@ class ClearStateStore implements StateStore {
...
  * @param stateStore - Replacement function for the built-in `stateStore`
+ * @param stateVerification - Pass in false to disable state parameter verification
  * @param installationStore - Interface to store and retrieve installation data from the database
  * @param authVersion - Can be either `v1` or `v2`. Determines which slack Oauth URL and method to use
  * @param logger - Pass in your own Logger if you don't want to use the built-in one
  * @param logLevel - Pass in the log level you want (ERROR, WARN, INFO, DEBUG). Default is INFO
  */
 export class InstallProvider {
-  public stateStore: StateStore;
+  public stateStore?: StateStore;
```
Comment (srajiang) [inline packages/oauth/src/index.ts:51]: When `stateVerification = false` a stateStore is not necessary anymore.

**Review by srajiang — COMMENTED** [review]
Please give feedback and thoughts, thank you!

File: packages/oauth/src/index.ts [hunk 2]
```diff
@@ -546,6 +579,7 @@ export interface CallbackOptions {
 export interface StateStore {
   // Returned Promise resolves for a string which can be used as an
   // OAuth state param.
+  // TODO: Revisit design. Does installOptions need to be encoded in state if metadata is static?
```
Comment (seratch) [inline packages/oauth/src/index.ts:582]: If I were you, I would add something like this:

TODO: With bolt-js built-in receivers, `InstallURLOptions` is always static. This means that metadata value is not so useful for customizing OAuth flows per user. If we enhance both this SDK and bolt-js to have a function that accepts request information (e.g., path, headers, query string) plus dynamically generate `InstallURLOptions` per request, developers can easily switch not only medatada but also redirect_uri, scopes, and user_scopes to support multiple installation patterns.

**Review by seratch — APPROVED** [review]
Left a comment on A TODO. Looks great to me. Squash commits when merging this 👋

**Pull request merged**